river/tree/hoeffding_tree_regressor.py

```diff
@@ -72,13 +82,25 @@ class HoeffdingTreeRegressor(BaseHoeffdingTree, base.Regressor):
     Hoeffding trees rely on Attribute Observer (AO) algorithms to monitor input features
     and perform splits. Nominal features can be easily dealt with, since the partitions
     are well-defined. Numerical features, however, require more sophisticated solutions.
-    Currently, only one AO is supported in `river` for regression trees:
+    Currently, three AO algorithms are supported in `river` for regression trees:
```

Comment:
The `notes` section seems now a good candidate for a page in the documentation. It is a nice chance to see how to extend documentation and direct the user from the implementation to this extra information. This can be done in the new PR for AOs.

Reply:
I agree. Thus, each AO/Splitter would have its own documentation and we would also provide a nice and high level comparison of them. 

I know it has been a long time now, but I still want to do a walkthrough in the tree module. Soon, we should have Stochastic Gradient Trees too, so our comparisons will be broader.